File: Data Structures & Algorithms/two-integer-sum-ii/submission-5.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int j = numbers.size() - 1;
        int i = 0;

        while (i < j) {
            int sum = numbers[j] + numbers[i];
            if (sum > target) {
                j--;
            } else if (sum < target) {
                i++;
            } else {
                return {i + 1, j + 1};
            }
        }

        return {};

    }    
};

File: Data Structures & Algorithms/is-anagram/submission-5.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:

        d = {};
        d1 = {};

        if len(s) != len(t):
            return False

        i = 0;
        while i < len(s):
            if s[i] not in d:
                d[s[i]] = 1
            else:
                d[s[i]] += 1
            
            if t[i] not in d1:
                d1[t[i]] = 1
            else:
                d1[t[i]] += 1
            i += 1
        
        return d == d1
        